### PR TITLE
Queries completed, formatted

### PR DESCRIPTION
**Pull request by Ervins-Trans**
Awaiting feedback, permission to merge

### CODE REVIEW

File: databases/week1/db_homework_week1.sql [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+WHERE
+    due_date IS NULL;
+--Find all the tasks that are marked as done
+SELECT
+    task.title AS 'DONE'
+FROM
+    task
+JOIN status ON status_id = task.status_id
+WHERE
+    status.name = 'DONE';
```
Comment (dsamourkasidis) [inline databases/week1/db_homework_week1.sql:20]: There are two problems with this:
1. If you look at the status table, the status name for a completed task is 'Done' not 'DONE'. 
2. When you join with the status table, there is no status_id column in the status table, but id. Instead this ```status_id=task.status_id``` should be ```status.id==task.status_id```

There is a simpler way to do this: If you notice, the 'Done' status_id is 3. So, it would be enough if you queried the task table where the task.status_id = 3

File: databases/week1/db_homework_week1.sql [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+--Find all the tasks that are not marked as done
+SELECT
+    task.title AS 'not DONE',
+    status.name AS 'actual status'
+FROM
+    task
+JOIN status ON status_id = task.status_id
+WHERE
+    status.name != 'DONE';
```
Comment (dsamourkasidis) [inline databases/week1/db_homework_week1.sql:29]: Same as previous comment with the 'DONE' tasks

File: databases/week1/db_homework_week1.sql [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+--Get all the tasks, sorted with the most recently created first
+SELECT
+    *
+FROM
+    task
+ORDER BY
+    -created;
```
Comment (dsamourkasidis) [inline databases/week1/db_homework_week1.sql:36]: Use DESC instead of -
```
ORDER BY
    created DESC;
```

File: databases/week1/db_homework_week1.sql [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+--Get the single most recently created task
+SELECT
+    *
+FROM
+    task
+ORDER BY
+    -created
```
Comment (dsamourkasidis) [inline databases/week1/db_homework_week1.sql:44]: same, use DESC instead

File: databases/week1/db_homework_week1.sql [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+WHERE
+    task.title LIKE '%database%' OR
+    task.description LIKE '%database%';
+--Get the title and status (as text) of all tasks
+SELECT
+    task.title,
+    status.name as 'text'
+FROM
+    task
+JOIN status ON task.status_id = status_id;
```
Comment (dsamourkasidis) [inline databases/week1/db_homework_week1.sql:62]: Again, pay attention to the status table columns -> it's not status_id it's id, so you should use status.id

File: databases/week1/db_homework_week1.sql [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+--Get the name of each status, along with a count of how many tasks have that status
+SELECT
+    name as 'status name',
+    count (task.id) as 'task count'
+FROM
+    status
+JOIN task ON status.id = task.status_id
```
Comment (dsamourkasidis) [inline databases/week1/db_homework_week1.sql:69]: Here you got it right :)

File: databases/week1/db_homework_week1.sql [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+GROUP BY
+    name
+ORDER BY
+    -count (task.id);
```
Comment (dsamourkasidis) [inline databases/week1/db_homework_week1.sql:82]: Use DESC instead

**Review by dsamourkasidis — COMMENTED** [review]
Very good work here, well done 👍 
Nice formatting, readable and clean :)
There are some issues you should fix, but they are very few and mostly just repeated

File: databases/week1/db_homework_week1.sql (hunk 3 above)
Comment (Ervins-Trans) [reply databases/week1/db_homework_week1.sql:36]: I will, but may i ask to elaborate, is "-" covering less cases, DESC being standard, more readable code, or why should i specifically use DESC ? Thanks in advance.

Comment (dsamourkasidis) [reply databases/week1/db_homework_week1.sql:36]: Absolutely, sorry for not elaborating initially :) 
So, while in this specific case where you want to sort by a datetime non-indexed column it is working as expected, there are cases where it would not work or not be optimal:

- If it was a numerical column it would work, but it doesn't work for text columns, try it with title for instance
- When you use an expression in the order by clause, [MySQL cannot use indexes](https://dev.mysql.com/doc/refman/8.0/en/order-by-optimization.html#:~:text=in%20some%20cases%2C%20mysql%20cannot%20use%20indexes%20to%20resolve%20the%20order%20by%2C%20although%20it%20may%20still%20use%20indexes%20to%20find%20the%20rows%20that%20match%20the%20where%20clause.%20examples%3A%20) to do the sorting (if the column is an index). In plain words, this means it will be slower!

So, basically it works perfectly fine here in this case, but there are cases where it could not, so to avoid having to think about it in the future, and since we already have the DESC keyword which does exactly that, I suggest you to use it instead :)
